pkgs/build-support/docker/default.nix

```diff
@@ -886,4 +888,70 @@ rec {
         done;
       '' + extraCommands;
     });
+
+  # This function builds a docker image that behaves like a nix-shell
+  buildNixShellImage =
+    # Use the environment of this derivation
+    { name
+    , drv
+    , tag ? null
+    # Extra packages available inside the image
+    , extraPackages ? []
+    # Extra bash commands to execute when the image is ran
+    , extraInit ? ""
+    }:
+    # Remove once https://github.com/NixOS/nixpkgs/pull/77952 is merged
+    let
+
+      finalDrv = drv.overrideAttrs (old: {
+        nativeBuildInputs = old.nativeBuildInputs or [] ++ extraPackages;
+      });
+
+      setupPackage = runCommandNoCC "setup" {
+        profile = ''
+          # This mimics part of nix-shell's initialization
+          source $stdenv/setup
+          set +e
+          eval "$shellHook"
+          cd /home
+        '' + extraInit;
```

Comment:
`extraInit` seems like a rather arbitrary term that doesn't have a well-understood purpose. Can't we remove it and rely on the user extending `shellHook` beforehand, or setting an appropriate `Cmd`?

`cd /home` breaks the `WorkingDir` config item.

Reply:
Yeah I'm removing `extraInit` in the next iteration. Can also be done with just `shellHook`.

`WorkingDir` can't be configured in the current version anyways

Reply:
With docker you can always configure any config attribute afterwards.

Reply:
Now using `WorkingDir` directly to specify this, instead of `cd` ing